addons/website_sale_slides/static/src/js/slides_course_join.js

```diff
@@ -0,0 +1,46 @@
+odoo.define('website_sale_slides.course.join.widget', function (require) {
+"use strict";
+
+var CourseJoinWidget = require('website_slides.course.join.widget').courseJoinWidget;
+
+CourseJoinWidget.include({
+    xmlDependencies: (CourseJoinWidget.prototype.xmlDependencies || []).concat(
+        ["/website_sale_slides/static/src/xml/channel_management.xml"]
+    ),
+    init: function (parent, options) {
+        this._super.apply(this, arguments);
+        this.productId = options.channel.productId || false;
+    },
+    //--------------------------------------------------------------------------
+    // Handlers
+    //--------------------------------------------------------------------------
+
+    /**
+     * @override
+     * @param {MouseEvent} ev
+     * @private
+     */
+    _onClickJoin: function (ev) {
+        this._super.apply(this, arguments);
+        if (this.channel.channelEnroll === 'payment') {
+            var url = _.str.sprintf('/shop/cart/update?product_id=%s&amp;express=1', this.productId);
+            if (this.publicUser) {
+                url = this._getLoginRedirectUrlSale(url);
+            } 
+            this.trigger_up('before_join_course_redirect', {url: url});
```

Comment:
This should not be needed anymore, right?

Reply:
Indeed, I forgot to change website_sale_slide, sorry.
It is done now.